cloud/EAP7-1769-Helm-cahrts-Configuring-Kubernetes-startup-probe.adoc

```diff
@@ -0,0 +1,137 @@
+
+Sometimes the pod could need more than 30 seconds to start, for example:
+
+The user has registered other liveness checks in their application code and to satisfy those conditions is needed more time.
+The management operations executed at runtime when the pod is started could also take some time to be applied, which delays the availability of the endpoint used by the default liveness probe.
+The node where the pod is running is slow.
+In such a case, users could be tempted to increase the `initialDelaySeconds` or the `failureThreshold` configuration, however increasing the `initialDelaySeconds` will delay a fixed value, which could compromise the expected fast response. Increasing the `failureThreshold` could also break the established conditions you have decided to check your application at runtime.
+
+The alternative to those issues is to have the ability to configure a Startup Probe. The Startup Probe can be used to delay the initial checks done by the liveness and readiness probes, allowing some configuration without compromising them.
+
+The goal of this RFE is to define the default initial configuration of a Startup Probe so users can tweak it when required.
+
+
+=== User Stories
+
+As a user, I want to delay the initial liveness and readiness check using an Startup probe.
+
+== Issue Metadata
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1769
+
+=== Affected Projects or Components
+
+* WildFly Helm Charts
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Startup probe configuration will be available in the Helm Chart.
+* The Helm Chart will add by default a Startup Probe to give three minutes as additional time before starting the Readiness and Liveness probes.
+* Users should be able to tweak all parameters of probes, which includes all timeouts, delays or type of probe.
+
+=== Changed requirements
+
+* N/A
+
+=== Non-Requirements
+
+* N/A
+
+=== Future Work
+
+* N/A
+
+== Backwards Compatibility
+
+* N/A
+
+=== Default Configuration
+
+* The proposal for the default Startup Probe configuration is the following:
++
+[source,yaml]
+startupProbe:
+  httpGet:
+    path: /health/live
+    port: admin
+  initialDelaySeconds: 5
+  failureThreshold: 36
+  periodSeconds: 5
+
+=== Importing Existing Configuration
+
+* N/A
+
+=== Deployments
+
+* N/A
+
+=== Interoperability
+
+* N/A
+
+== Implementation Plan
+
+* N/A
+
+== Admin Clients
+
+* N/A
+
+== Security Considerations
+
+* N/A
+
+[[test_plan]]
+== Test Plan
+
+* Test plan will be delivered by a professional Quality Engineer with subject matter expertise.
+
+== Community Documentation
+
+The Helm Chart docs will include a description of the uses of the Startup Probe.
+
+== Release Note Content
+
+Added the ability to configure a Startup Probe using WildFly Helm Charts.
```

Comment:
```suggestion
Increased default timeouts for startup, readiness and liveness probes and added the ability to configure a these probes using WildFly Helm Charts.
```

Reply:
Why `Increased default timeouts for startup, readiness and liveness probes` if we are not adding any specific configuration for readiness and liveness?

Reply:
:thinking:  poor choice of words on my side (also caused by inconsistencies in productized values) 
```suggestion
Increased default timeouts for startup probe and added the ability to configure a all probes (startup, readiness and liveness)  using WildFly Helm Charts.
```